ludwig/features/binary_feature.py

```diff
@@ -72,7 +72,7 @@ def forward(self, v: Union[List[str], List[torch.Tensor], torch.Tensor]):
         if self.should_lower:
             v = [s.lower() for s in v]
         indices = [self.str2bool.get(s, False) for s in v]
-        return torch.tensor(indices, dtype=torch.bool)
+        return torch.tensor(indices, dtype=torch.float32)
```

Comment:
Not sure I understand why this being done. It is also at odds with line 69 above, which explicitly casts to a bool type.

Reply:
Upon saving and loading the torchscript module, it seems to raise an error on binary-only features with the following message:

```
...
E           Traceback of TorchScript, original code (most recent call last):
E             File "/Users/geoffreyangus/repositories/predibase/ludwig/ludwig/models/inference.py", line 71, in forward
E                           feature_name = get_name_from_module_dict_key(module_dict_key)
E                           preproc_inputs[feature_name] = preproc(inputs[feature_name])
E                       outputs = self.model(preproc_inputs)
E                                 ~~~~~~~~~~ <--- HERE
E               
E                       predictions: Dict[str, Dict[str, torch.Tensor]] = {}
E           /usr/local/lib/python3.8/site-packages/torch/nn/modules/linear.py(103): forward
...
E           RuntimeError: expected scalar type Float but found Bool
```

This change probably doesn't belong in its own PR, though. I'll put up another one with the change and a regression test. Good catch!

Reply:
Put this change+regression test in a new PR here: #2103